### PR TITLE
Add storage cluster to stress environments. Update federated auth documentation.

### PR DESCRIPTION
**Pull request by benbp**
I have built out a new default stress cluster in the storage partner team subscription that they can use. Adding it to the config defaults so they can use it out of the box.

Also updating the documentation to reflect the changes made in https://github.com/Azure/azure-sdk-tools/pull/8278 to support workload identity as the auth mode for the stress clusters.

### CODE REVIEW

File: tools/stress-cluster/chaos/README.md
Comment (richardpark-msft) [inline tools/stress-cluster/chaos/README.md:195]: I think we should just eliminate any mention of AzureCliCredential. A bit later, in this doc, you cover how they could use the Azure CLI - that seems like a reasonable spot to document the possibility.

We definitely don't want them to login to the CLI just to get their normal test running (ie, mentioning it here confuses things).

File: tools/stress-cluster/chaos/README.md
Comment (richardpark-msft) [inline tools/stress-cluster/chaos/README.md:203]: I have a PR to make the .env file sourceable (sometimes the unquoted values mess up bash). 

https://github.com/Azure/azure-sdk-tools/pull/7830

We can update this doc after that comes in, or perhaps you are quoting everything now?

File: tools/stress-cluster/chaos/README.md
Comment (richardpark-msft) [inline tools/stress-cluster/chaos/README.md:211]:
```suggestion
When a new namespace is created in the stress cluster, a [federated identity](https://learn.microsoft.com/en-us/graph/api/resources/federatedidentitycredentials-overview?view=graph-rest-1.0) specific to that namespace is created against a pool of managed identities by the stress watcher service (there is a hard limit of 20 federated identities per managed identity). When a namespace is deleted, the corresponding federated identity is also deleted.
```

File: tools/stress-cluster/chaos/README.md
Comment (richardpark-msft) [inline tools/stress-cluster/chaos/README.md:246]:
```suggestion
- `AZURE_TENANT_ID` - The Azure tenant id the stress test will authenticate to. Set by AKS.
```

File: tools/stress-cluster/chaos/README.md
Comment (richardpark-msft) [inline tools/stress-cluster/chaos/README.md:245]:
```suggestion
- `AZURE_SUBSCRIPTION_ID` - The Azure subscription id the stress test will authenticate and deploy resources to.
```

File: tools/stress-cluster/chaos/README.md
Comment (richardpark-msft) [inline tools/stress-cluster/chaos/README.md:247]:
```suggestion
- `AZURE_CLIENT_ID` - The Entra principal the stress test will authenticate as. Set by AKS.
```

File: tools/stress-cluster/chaos/README.md
Comment (richardpark-msft) [inline tools/stress-cluster/chaos/README.md:248]:
```suggestion
- `AZURE_FEDERATED_TOKEN_FILE` - The path to the federated identity token that can be used to login with Azure CLI or the Identity SDK. Set by AKS.
```

**Review by richardpark-msft — APPROVED** [review]

File: tools/stress-cluster/chaos/README.md
Comment (benbp) [reply tools/stress-cluster/chaos/README.md:195]: Fair point, and I'm considering not even mentioning `WorkloadIdentityCredential` either. We should just say in code, use DAC, and in a script, use az cli login?

File: tools/stress-cluster/chaos/README.md
Comment (benbp) [reply tools/stress-cluster/chaos/README.md:203]: Approved that one, yeah we can rebase after merge and get that one in.

File: tools/stress-cluster/chaos/README.md
Comment (richardpark-msft) [reply tools/stress-cluster/chaos/README.md:195]: I was okay with that, mostly because I it helps in identifying which credential was supposed to work. Went through this a little with @LibbaLawrence where we weren't sure which credential failure was the important one.

File: tools/stress-cluster/chaos/README.md
Comment (benbp) [reply tools/stress-cluster/chaos/README.md:211]: I said `a` stress cluster because we have many. But if you think `the` still makes more sense I can take this change.

**Pull request merged**
